staging/src/k8s.io/apiserver/pkg/storage/etcd3/compact.go

```diff
@@ -160,3 +163,16 @@ func compact(ctx context.Context, client *clientv3.Client, t, rev int64) (int64,
 	klog.V(4).Infof("etcd: compacted rev (%d), endpoints (%v)", rev, client.Endpoints())
 	return curTime, curRev, nil
 }
+
+func CompactTestStorage(etcdClient *clientv3.Client) testing.Compaction {
```

Comment:
Didn't find any other way to make this available in `testing` package while still use real `compact` function. Compaction should be taken out of Storage IMHO.